Detectors/Calibration/include/DetectorsCalibration/MeanVertexData.h

```diff
@@ -0,0 +1,231 @@
+    histoY.clear();
+    histoZ.clear();
+  }
+
+MeanVertexData(bool buseFit, int nbX, float rX, int nbY, float rY, int nbZ, float rZ) :
+  useFit(buseFit), nbinsX(nbX), rangeX(rX), v2BinX(0), nbinsY(nbY), rangeY(rY), v2BinY(0),
+    nbinsZ(nbZ), rangeZ(rZ), v2BinZ(0) 
+  {
+    //    if (useFit) {
+      if (rX <= 0. || nbX < 1 || rY <= 0. || nbY < 1 || rZ <= 0. || nbZ < 1) {
+	throw std::runtime_error("Wrong initialization of the histogram");
+      }
+      v2BinX = nbinsX / (2 * rangeX);
+      v2BinY = nbinsY / (2 * rangeY);
+      v2BinZ = nbinsZ / (2 * rangeZ);
+      histoX.resize(nbinsX, 0.);
+      histoY.resize(nbinsY, 0.);
+      histoZ.resize(nbinsZ, 0.);                           
+    }
+  //}
+
+//_____________________________________________
+MeanVertexData(MeanVertexData&& other) {
+
+    // move constructor
+    
+    useFit = other.useFit;
+    nbinsX = other.nbinsX;
+    rangeX = other.rangeX;
+    nbinsY = other.nbinsY;
+    rangeY = other.rangeY;
+    nbinsZ = other.nbinsZ;
+    rangeZ = other.rangeZ;
+    v2BinX = other.v2BinX;
+    v2BinY = other.v2BinY;
+    v2BinZ = other.v2BinZ;
+    histoX = other.histoX;
+    histoY = other.histoY;
+    histoZ = other.histoZ;
+    other.useFit = false;
+    other.nbinsX = 1000;
+    other.rangeX = 10.f;
+    other.nbinsY = 1000;
+    other.rangeY = 10.f;
+    other.nbinsZ = 1000;
+    other.rangeZ = 10.f;
+    other.v2BinX = nbinsX / (2 * rangeX);
+    other.v2BinY = nbinsY / (2 * rangeY);
+    other.v2BinZ = nbinsZ / (2 * rangeZ);
+    other.histoX.clear();
+    other.histoY.clear();
+    other.histoZ.clear();
+}
+
+//_____________________________________________
+MeanVertexData(const MeanVertexData& other) {
+
+    // copy constructor
+    
+    useFit = other.useFit;
+    nbinsX = other.nbinsX;
+    rangeX = other.rangeX;
+    nbinsY = other.nbinsY;
+    rangeY = other.rangeY;
+    nbinsZ = other.nbinsZ;
+    rangeZ = other.rangeZ;
+    v2BinX = other.v2BinX;
+    v2BinY = other.v2BinY;
+    v2BinZ = other.v2BinZ;
+    histoX = other.histoX;
+    histoY = other.histoY;
+    histoZ = other.histoZ;
+}
+
+//_____________________________________________
+MeanVertexData& operator = (MeanVertexData& other) {
+
+    // assignment operator
+    
+    useFit = other.useFit;
+    nbinsX = other.nbinsX;
+    rangeX = other.rangeX;
+    nbinsY = other.nbinsY;
+    rangeY = other.rangeY;
+    nbinsZ = other.nbinsZ;
+    rangeZ = other.rangeZ;
+    v2BinX = other.v2BinX;
+    v2BinY = other.v2BinY;
+    v2BinZ = other.v2BinZ;
+    histoX = other.histoX;
+    histoY = other.histoY;
+    histoZ = other.histoZ;
+    other.useFit = false;
+    other.nbinsX = 1000;
+    other.rangeX = 10.f;
+    other.nbinsY = 1000;
+    other.rangeY = 10.f;
+    other.nbinsZ = 1000;
+    other.rangeZ = 10.f;
+    other.v2BinX = nbinsX / (2 * rangeX);
+    other.v2BinY = nbinsY / (2 * rangeY);
+    other.v2BinZ = nbinsZ / (2 * rangeZ);
+    other.histoX.clear();
+    other.histoY.clear();
+    other.histoZ.clear();
```

Comment:
I don't understand, why do you modify ``other``? If to reset it, one should rely on default assignment operator and add separate ``clear`` method.

Reply:
I thought that this was needed to call std::move correctly. 
I now use the default implementations, so I removed it.